docs/tutorials/creating-a-landsat-stac.ipynb

```diff
@@ -49,743 +63,521 @@
+    "Landsat metadata is contained in an `MTL` file that comes in either `.txt` or `.xml` formats.  We'll rely on the XML version since it is more consistently available.  This will require that we provide some facility for parsing the XML into a more usable format:"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 3,
    "metadata": {},
    "outputs": [],
    "source": [
-    "lat, lon = 39.9526, -75.1652\n",
-    "location_name = \"Philly\"\n",
-    "filter_year = \"2020\""
+    "# Taken from https://stackoverflow.com/questions/2148119/how-to-convert-an-xml-string-to-a-dictionary\n",
+    "from xml.etree import cElementTree as ElementTree\n",
+    "\n",
+    "\n",
+    "class XmlListConfig(list):\n",
+    "    def __init__(self, aList):\n",
+    "        for element in aList:\n",
+    "            if element:\n",
+    "                if len(element) == 1 or element[0].tag != element[1].tag:\n",
+    "                    self.append(XmlDictConfig(element))\n",
+    "                elif element[0].tag == element[1].tag:\n",
+    "                    self.append(XmlListConfig(element))\n",
+    "            elif element.text:\n",
+    "                text = element.text.strip()\n",
+    "                if text:\n",
+    "                    self.append(text)\n",
+    "\n",
+    "\n",
+    "class XmlDictConfig(dict):\n",
+    "    def __init__(self, parent_element):\n",
+    "        if parent_element.items():\n",
+    "            self.update(dict(parent_element.items()))\n",
+    "        for element in parent_element:\n",
+    "            if element:\n",
+    "                if len(element) == 1 or element[0].tag != element[1].tag:\n",
+    "                    aDict = XmlDictConfig(element)\n",
+    "                else:\n",
+    "                    aDict = {element[0].tag: XmlListConfig(element)}\n",
+    "                if element.items():\n",
+    "                    aDict.update(dict(element.items()))\n",
+    "                self.update({element.tag: aDict})\n",
+    "            elif element.items():\n",
+    "                self.update({element.tag: dict(element.items())})\n",
+    "            else:\n",
+    "                self.update({element.tag: element.text})"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We'll use the coordinates and the year to filter out any unwanted scenes:"
+    "We can then use these classes to get the MTL file for our scene. Notice we use pystac.STAC_IO.read_text - this is the method that PySTAC uses to read text as it crawls a STAC. It can read from the local filesystem or HTTP/HTTPS by default. Also, it can be extended to read from other sources such as cloud providers—[see the documentation here](https://pystac.readthedocs.io/en/latest/concepts.html#using-stac-io). For now we'll use it directly as an easy way to read a text file from an HTTPS source."
```

Comment:
```suggestion
    "We can then use these classes to get the MTL file for our scene. Notice we use `pystac.STAC_IO.read_text` - this is the method that PySTAC uses to read text as it crawls a STAC. It can read from the local filesystem or HTTP/HTTPS by default. Also, it can be extended to read from other sources such as cloud providers—[see the documentation here](https://pystac.readthedocs.io/en/latest/concepts.html#using-stac-io). For now we'll use it directly as an easy way to read a text file from an HTTPS source."
```